tmt/steps/discover/shell.py

```diff
@@ -288,7 +298,10 @@ def fetch_remote_repository(
             self.run(Command('git', 'checkout', '-f', str(ref)), cwd=testdir)
 
         # Remove .git so that it's not copied to the SUT
-        shutil.rmtree(os.path.join(testdir, '.git'))
+        # if 'sync-repo' option is not specified
+        requires_git = self.opt('sync-repo')
```

Comment:
I guess we want to support enabling the repo syncing from discover config as well:

```yaml
discover:
    how: shell
    url: https://github.com/teemtee/tmt
    sync-repo: true
```

So we should use `self.get()` here instead.

Reply:
Hm, 'discover fmf' is using `self.opt` as well, so we should change that too (IMO we should have this in sync).

Reply:
Agreed, let's fix that one here as well.

Reply:
Currently [tmt/steps/discover/fmf.py](https://github.com/teemtee/tmt/blob/main/tmt/steps/discover/fmf.py) doesn't use `tmt.utils.field`, how about fixing it in a separate PR?

Reply:
I'd recommend a dedicated PR for such a change, it tends to have various side-effects, i.e. it would introduce changes unrelated to your goal in this PR.

Reply:
Makes sense, let's cover that in a separate PR.

Reply:
Sure, will file a separate PR after this one is merged.